### DISCUSSION TITLE
convert cannot display image on /dev/fb0

### DISCUSSION THREAD

**Discussion by robinmordasiewicz** [ImageMagick/ImageMagick#4196] [Help]
### Description
images displayed on the Framebuffer using convert are distorted

### Steps to Reproduce
convert -resize 1920x1080 -background black -gravity center -extent 1920x1080 /home/pi/Pictures/default_background.png bgra:/dev/fb0

### System Configuration
Raspberry Pi4

- ImageMagick version: ImageMagick 6.9.10-23 Q16 arm 20190101

**Comment by snibgo** [comment]
Your command options are in the wrong order. They should be: read the image, process it, write it, like this:
```
convert /home/pi/Pictures/default_background.png -resize 1920x1080 -background black -gravity center -extent 1920x1080 bgra:/dev/fb0
```
That may not solve your problem. I know nothing about Raspberry Pi.

It may help if you explain what you mean by "distorted".

**Comment by robinmordasiewicz** [comment]
Attached is a screenshot of the image as displayed with the command below.
A number of weeks ago convert stopped working.


convert /home/pi/Pictures/default_background.png -resize 1920x1080 -background black -gravity center -extent 1920x1080 bgra:/dev/fb0

[cid:image001.png@01D6B219.7E7435C0]

**Comment by snibgo** [comment]
I can't see a screenshot.

**Comment by urban-warrior** [comment]
You may need to add `-depth 8` to your command-line.  Also, make sure you have write permission to the framebuffer.
